### PR TITLE
Forward raw transactions to OP sequencer

### PR DESCRIPTION
**Pull request by cjeva10**
#3797 @merklefruit

Please ignore the earlier PR that had some merge conflicts.

### CODE REVIEW

**Comment by merklefruit** [comment]
Thank you for the PR!

Something still looks off in terms of commits, will either have to rebase or update the upstream before taking a look :)

**Comment by cjeva10** [comment]
Gotcha - please let me know if there's something I can do to fix.

**Comment by rkrasiuk** [comment]
@cjeva10 if `clabby/op-reth` has been rebased since you created your branch, you might need to do the same

File: crates/rpc/rpc/src/eth/api/transactions.rs [hunk 1]
```diff
@@ -390,6 +392,26 @@ where
...
+            }];
+
+            let req = Request::builder()
+                .method(Method::POST)
+                .uri("https://mainnet-sequencer.optimism.io")
+                .header("Content-type", "application/json")
+                .body(Body::from(body));
+
+            let res = client.request(req).await?;
+        }
```
Comment (merklefruit) [inline crates/rpc/rpc/src/eth/api/transactions.rs:413]: The `chain_id == 10` check is not going to work for all chains based on the OP stack, which is our goal here.

What would probably be a good alternative is checking if the "sequencer http endpoint" CLI flag is set instead, which is why this ticket was marked as blocked by https://github.com/paradigmxyz/reth/issues/3794

As this flag has been added in #33 let's wait for it to be upstreamed and then will come back to this @cjeva10

**Comment by merklefruit** [comment]
Hey @cjeva10, I rebased your changes on another branch here: #41 
I also plugged in the sequencer http endpoint flag so users of the node can customize this behavior. 

Thank you again for the contribution, and feel free to give a review of my PR if I missed anything :)

**Pull request closed without merging**